packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/index.js

```diff
@@ -1,136 +1,35 @@
 import React from 'react';
+import { useUserPermissions, EmptyState } from 'strapi-helper-plugin';
 import { useIntl } from 'react-intl';
-import { BaselineAlignment, ModalConfirm, EmptyList, ListButton } from 'strapi-helper-plugin';
-import { Header, List } from '@buffetjs/custom';
-import { Button, Text } from '@buffetjs/core';
-import { Plus } from '@buffetjs/icons';
-import ModalEdit from '../../components/ModalEdit';
-import { LocaleRow } from '../../components';
-import { useLocales } from '../../hooks';
+import LocaleSettingsPage from './LocaleSettingsPage';
+import i18nPermissions from '../../permissions';
 import { getTrad } from '../../utils';
-import useDeleteLocale from '../../hooks/useDeleteLocale';
-import useEditLocale from '../../hooks/useEditLocale';
 
-// Fake permissions
-const canCreate = true;
-const canDelete = true;
-const canEdit = true;
-
-const LocaleSettingsPage = () => {
-  const {
-    isDeleting,
-    isDeleteModalOpen,
-    deleteLocale,
-    showDeleteModal,
-    hideDeleteModal,
-  } = useDeleteLocale();
-
-  const { isEditing, isEditModalOpen, editLocale, showEditModal, hideEditModal } = useEditLocale();
+const ProtectedLocaleSettingsPage = () => {
   const { formatMessage } = useIntl();
-  const { locales, isLoading } = useLocales();
-
-  const handleDelete = canDelete ? showDeleteModal : undefined;
-  const handleEdit = canEdit ? showEditModal : undefined;
-
-  const actions = [
-    {
-      label: formatMessage({ id: getTrad('Settings.list.actions.add') }),
-      onClick: () => console.log('add locale'),
-      color: 'primary',
-      type: 'button',
-      icon: true,
-      Component: props => {
-        if (canCreate) {
-          return <Button {...props} />;
-        }
-
-        return null;
-      },
-      style: {
-        paddingLeft: 15,
-        paddingRight: 15,
-      },
-    },
-  ];
-
-  const listTitle = isLoading
-    ? null
-    : formatMessage(
-        {
-          id: getTrad(`Settings.locales.list.title${locales.length > 1 ? '.plural' : '.singular'}`),
-        },
-        { number: locales.length }
-      );
-
-  return (
-    <>
-      <Header
-        title={{
-          label: formatMessage({ id: getTrad('plugin.name') }),
-        }}
-        content={formatMessage({ id: getTrad('Settings.list.description') })}
-        actions={actions}
+  const {
+    isLoading,
+    allowedActions: { canRead, canUpdate, canCreate, canDelete },
+  } = useUserPermissions(i18nPermissions);
+
+  if (isLoading) {
+    return (
+      <div>
+        <p>{formatMessage({ id: getTrad('Settings.permissions.loading') })}</p>
+      </div>
+    );
+  }
+
+  if (!canRead) {
```

Comment:
When the user is not allowed to read the content, I'm displaying a message. Is this something we want or should I just redirect the user?

Reply:
Usually, we simply don't display the component.